libs/gui/src/CAboutBoxBase.cpp

```diff
@@ -0,0 +1,89 @@
+/* +------------------------------------------------------------------------+
+   |                     Mobile Robot Programming Toolkit (MRPT)            |
+   |                          http://www.mrpt.org/                          |
+   |                                                                        |
+   | Copyright (c) 2005-2017, Individual contributors, see AUTHORS file     |
+   | See: http://www.mrpt.org/Authors - All rights reserved.                |
+   | Released under BSD License. See details in http://www.mrpt.org/License |
+   +------------------------------------------------------------------------+ */
+#include "CAboutBoxBase.h"
+#include <mrpt/config.h>
+#include <mrpt/system/os.h>
+
+using namespace mrpt;
+using namespace mrpt::system;
+
+CAboutBoxBase::CAboutBoxBase(
+	const std::string& appName, const std::string& additionalInfo,
+	const bool showStandardInfo)
+	: m_appName(appName),
+	  m_additionalInfo(additionalInfo),
+	  m_showStandardInfo(showStandardInfo)
+{
+}
+
+CAboutBoxBase::~CAboutBoxBase() {}
+std::string CAboutBoxBase::MRPTBuildVersion() const
+{
+	return "Build: " + MRPT_getVersion() +
+		   mrpt::system::MRPT_getCompilationDate();
+}
+
+std::string CAboutBoxBase::tutorial() const
+{
+	return "Up to date documentation and tutorials are maintained at the MRPT "
+		   "website:\n\nhttp://www.mrpt.org/\n\n\n\n";
+}
+
+std::string CAboutBoxBase::license() const
+{
+	return mrpt::system::getMRPTLicense();
+}
+
+std::string CAboutBoxBase::information(
+	const std::string& guiLibName, const std::string& guiLibVersion) const
+{
+	std::string str = m_appName + "\n";
+	str +=
+		"----------------------------------\n"
+		"Part of the MRPT project.\n"
+		"For bug reports and source code, visit:"
+		" https:://github.com/MRPT/mrpt \n\n";
+	;
+	if (!m_additionalInfo.empty())
+	{
+		str += m_additionalInfo + "\n";
+	}
+	str += "MRPT version:           " + MRPT_getVersion() + "\n";
+	str += "MRPT source timestamp:  " + MRPT_getCompilationDate() + "\n";
+
+	if (m_showStandardInfo)
+	{
+		str += "Eigen version:          ";
+		str += EIGEN_WORLD_VERSION + ".";
+		str += EIGEN_MAJOR_VERSION + ".";
+		str += EIGEN_MINOR_VERSION + "\n";
+
+		str += guiLibName + " version:      " + guiLibVersion;
+#if defined(__WXMSW__)
+		str += "-Windows";
+#elif defined(__UNIX__)
+		str += "-Linux";
+#endif
```

Comment:
Should we also add a Qt version line here?

Reply:
I use "guiLibName" for this, which defined for Qt in the constructor CAboutBoxQt and for wx in the  CAboutBox::OnInit.